frog/imports/api/generateReactiveFn.js

```diff
@@ -187,7 +187,14 @@ export class Doc {
       'li',
       id
     );
-
+    if (LIData.liType === 'li-activity') {
+      // In this case the activity also needs to be duplicated
+      const acId = uuid();
+      const activityPointer = connection.get('rz', acId);
+      // New instance starts with empty data
+      activityPointer.create({});
```

Comment:
this should be activityTypesObj[acId].dataStructure || {}

Reply:
so the need for a specific duplication mechanism is exactly why we have the option of having a duplicate function in the liType... so can you move this logic to li-activity, adding a duplicate function to the main export?

Reply:
also, should a new instance start with empty data even if there is already a lot of data? that's not my intuition of cloning - if I have a rich text with lots of text, and clone it, I get two lis with lots of text... cloning a ck-board should be the same?

Reply:
Where should the instance get the new data from? (This is for creating a new instance when a new individual comes)

Reply:
If we export it into the `li-activity` we won't be able to assign the `id` of the activity to the `rz` field of the LI.